data/gui/widget/combobox_default.cfg

```
@@ -0,0 +1,277 @@
+#textdomain wesnoth-lib
+###
+### Definition of a combobox.
+### A widget with a text box and a list. Selecting an element from the list
+### set the value of the text box to that item of the list. The user can also
+### manually enter a value in the text box.
+###
+
+#define _GUI_DRAW_BACKGROUND COLOR
+	[rectangle]
+		x = 0
+		y = 0
+		w = "(width)"
+		h = "(height)"
+
+		fill_color = {COLOR}
+
+	[/rectangle]
+#enddef
+
+#define _GUI_TEXTBOX_BACKGROUND_ENABLED
+	{_GUI_DRAW_BACKGROUND ({GUI__BACKGROUND_COLOR_ENABLED})}
+#enddef
+
+#define _GUI_TEXTBOX_BACKGROUND_DISABLED
+	{_GUI_DRAW_BACKGROUND ({GUI__BACKGROUND_COLOR_DISABLED})}
+#enddef
+
+#define _GUI_DRAW_BORDER COLOR
+	[rectangle]
+		x = 0
+		y = 0
+		w = "(width)"
+		h = "(height)"
+
+		border_thickness = 1
+		border_color = {COLOR}
+
+	[/rectangle]
+#enddef
+
+#define _GUI_DRAW_TEXT SIZE COLOR
+#arg FONT_FAMILY
+#endarg
+	[rectangle]
+		x = "(text_x_offset + selection_offset)"
+		y = "(text_y_offset)"
+		w = "(selection_width)"
+		h = "(text_font_height)"
+
+		border_thickness = 0
+		fill_color = "21, 53, 80, 255"
+	[/rectangle]
+
+	[text]
+		x = "(text_x_offset)"
+		y = "(text_y_offset)"
+		w = "(text_width)"
+		h = "(text_height)"
+		maximum_width = "(text_maximum_width)"
+		font_size = {SIZE}
+		font_family = {FONT_FAMILY}
+		color = {COLOR}
+		text = "(text)"
+	[/text]
+#enddef
+
+#define __DISABLED_COLOR_PREPROCESSOR_WORKAROUND
+128, 128, 128, 255 #enddef
+
+#define _GUI_DRAW_TEXT_OR_HINT SIZE COLOR
+#arg FONT_FAMILY
+#endarg
+	[rectangle]
+		x = "(text_x_offset + selection_offset)"
+		y = "(text_y_offset)"
+		w = "(selection_width)"
+		h = "(text_font_height)"
+
+		border_thickness = 0
+		fill_color = "21, 53, 80, 255"
+	[/rectangle]
+
+	[text]
+		x = "(text_x_offset)"
+		y = "(text_y_offset)"
+		w = "(text_width)"
+		h = "(text_height)"
+		maximum_width = "(text_maximum_width)"
+
+		font_size = "(
+			if(text = '' and hint_text != '', hint_size, reg_size) where
+				hint_size = {GUI_FONT_SIZE_SMALL},
+				reg_size = {SIZE}
+			)"
+		font_family = {FONT_FAMILY}
+
+		color = "(
+			if(text = '' and hint_text != '', hint_color, reg_color) where
+				hint_color = [{__DISABLED_COLOR_PREPROCESSOR_WORKAROUND}],
```

```suggestion
				hint_color = ["{GUI__FONT_COLOR_DISABLED__DEFAULT}"],
```
And `__DISABLED_COLOR_PREPROCESSOR_WORKAROUND` define above should be removed.